src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -5,77 +5,39 @@
 using System;
 using System.Collections.Immutable;
 using System.Linq;
-using ILLink.Shared;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace ILLink.RoslynAnalyzer
 {
+	[AddSupportedDiagnostic ("IL3000", "AvoidAssemblyLocationInSingleFile",
+		Category = DiagnosticCategory.SingleFile,
+		HelpLinkURI = "https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/il3000")]
+	[AddSupportedDiagnostic ("IL3001", "AvoidAssemblyGetFilesInSingleFile",
+		Category = DiagnosticCategory.SingleFile,
+		HelpLinkURI = "https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/il3001")]
+	[AddSupportedDiagnostic ("IL3002", "RequiresAssemblyFiles",
+		Category = DiagnosticCategory.SingleFile,
+		HelpLinkURI = "https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/il3002")]
+	[AddSupportedDiagnostic ("IL3003", "RequiresAttributeMismatch")]
```

Comment:
This will default to category trimming and is still a SingleFile warning

Reply:
I simply put the category that it has today:

https://github.com/mono/linker/blob/2174dc1bf8217ab8f5b77b0fcc0ab0cbf1973a85/src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs#L64

Do you think we should change it then?

Reply:
Yikes, me putting comments for my own mistakes 😂
Yes, this is not a trimming warning is a mismatch between RequiresAssemblyFiles so it should have the SingleFile category